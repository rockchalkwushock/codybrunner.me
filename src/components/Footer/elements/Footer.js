import styled from 'styled-components'

const Footer = styled.footer`
  align-items: center;
  display: flex;
  flex-direction: column;
  font-size: 80%;
  justify-content: center;
  padding: 1em;

  @media (min-width: 736px) {
    flex-direction: row;
    font-size: 100%;
    justify-content: space-between;
  }
`

Footer.displayName = 'Footer'

export default Footer
