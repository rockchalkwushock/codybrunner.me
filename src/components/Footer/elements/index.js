import styled from 'styled-components'

const Footer = styled.footer`
  background-color: ${({ theme }) => theme.colors.secondary};
  padding: 1.5rem 0;
  width: 100%;
`

const FooterList = styled.ul`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  list-style: none;
  margin: ${({ theme }) => theme.reset};
  padding: ${({ theme }) => theme.reset};
`

const FooterItem = styled.li`
  margin: ${({ theme }) => theme.reset};
  padding: 0.25rem 0;
`

export { Footer, FooterItem, FooterList }
