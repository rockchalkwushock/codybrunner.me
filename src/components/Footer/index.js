import React from 'react'
import { shape, string } from 'prop-types'

import { Container, Footer } from './elements'
import Link from '../Link'

const SiteFooter = ({ buildTime, copyright, links, siteUrl }) => {
  const { creativeCommons, gatsby, now, src } = links
  return (
    <Footer>
      <Container left>
        <Link ext href={src.href} text={`Last build: ${buildTime}`} />
        <Link ext href={siteUrl} text={copyright} />
        <Link ext href={creativeCommons.href} text={creativeCommons.text} />
      </Container>
      <Container>
        <Link ext href={gatsby.href} text={gatsby.text} />
        <Link ext href={now.href} text={now.text} />
      </Container>
    </Footer>
  )
}

SiteFooter.propTypes = {
  builtTime: string,
  copyright: string.isRequired,
  links: shape({
    creativeCommons: shape({
      href: string.isRequired,
      text: string.isRequired
    }),
    gatsby: shape({
      href: string.isRequired,
      text: string.isRequired
    }),
    now: shape({
      href: string.isRequired,
      text: string.isRequired
    }),
    src: shape({
      href: string.isRequired
    })
  }),
  siteUrl: string.isRequired
}

export default SiteFooter
