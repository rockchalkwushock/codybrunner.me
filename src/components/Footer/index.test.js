import React from 'react'
import { shallow } from 'enzyme'

import Footer from './'

const { siteMetadata } = require('../../../gatsby-config')

describe('Component: <Footer />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <Footer
        buildTime="02-03-99"
        copyright="Â© Turd"
        links={siteMetadata.links}
        siteUrl="turdferguson.com"
      />
    )
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
