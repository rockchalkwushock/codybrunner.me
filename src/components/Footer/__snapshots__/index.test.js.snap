// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Footer /> should render correctly 1`] = `
<Footer>
  <Container
    left={true}
  >
    <SiteLink
      ext={true}
      href="https://github.com/rockchalkwushock/codybrunner.rocks"
      text="Last build: 02-03-99"
    />
    <SiteLink
      ext={true}
      href="turdferguson.com"
      text="Â© Turd"
    />
    <SiteLink
      ext={true}
      href="https://creativecommons.org/licenses/by-nc-sa/4.0/"
      text="All Rights Reserved."
    />
  </Container>
  <Container>
    <SiteLink
      ext={true}
      href="https://www.styled-components.com"
      text="Styled with Styled-Components 3.1"
    />
    <SiteLink
      ext={true}
      href="https://gatsbyjs.org"
      text="Powered by Gatsby 1.9"
    />
    <SiteLink
      ext={true}
      href="https://zeit.co/now"
      text="Hosted on Now"
    />
  </Container>
</Footer>
`;
